### PR TITLE
feat(@angular/cli): add ng4 option to ng new

### PR DESCRIPTION
**Pull request by hansl**
This allows to create a project with Angular 4 in the template.
Also fixes a few issues with ng4.

### CODE REVIEW

File: tests/e2e/setup/500-create-project.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function() {
   } else {
     // Otherwise create a project from scratch.
     createProject = Promise.resolve()
-      .then(() => ng('new', 'test-project', '--skip-npm'))
+      .then(() => ng('new', 'test-project', '--skip-npm', argv['ng4'] ? '--ng4' : '--'))
```
Comment (filipesilva) [inline tests/e2e/setup/500-create-project.ts]: Nit: `'--'` -> `''`

Comment (hansl) [inline tests/e2e/setup/500-create-project.ts]: Done.

**Comment by Splaktar** [comment]
Should this be `ngnext` instead of being hardcoded to v4.x (ng4)? I imagine that the need for generating projects with the latest (alpha/beta) Angular will continue to exist after ng4 is GA (i.e. ng5-alpha).

**Comment by fmalcher** [comment]
Are there any plans to remove the ng4 option again once Angular 4 is released?
@hansl @filipesilva

**Comment by filipesilva** [comment]
@Splaktar @fmalcher we're going to do something along those lines, yes.

**Pull request merged**